integration-cli/docker_cli_build_test.go

```diff
@@ -5660,3 +5660,434 @@ func (s *DockerSuite) TestBuildNullStringInAddCopyVolume(c *check.C) {
 	_, err = buildImageFromContext(name, ctx, true)
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestBuildBuildTimeArg(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+	args := []string{
+		"--build-arg", fmt.Sprintf("%s=%s", envKey, envVal),
+	}
+	dockerfile := fmt.Sprintf(`FROM busybox
+		ARG %s=
```

Comment:
That's correct
On Thu, Sep 10, 2015 at 12:18 PM Madhav Puri <notifications@github.com>
wrote:

> In integration-cli/docker_cli_build_test.go
> <https://github.com/docker/docker/pull/15182#discussion_r39179735>:
>
> > @@ -5660,3 +5660,434 @@ func (s *DockerSuite) TestBuildNullStringInAddCopyVolume(c *check.C) {
> >  	_, err = buildImageFromContext(name, ctx, true)
> >  	c.Assert(err, check.IsNil)
> >  }
> > +
> > +func (s *DockerSuite) TestBuildBuildTimeArg(c *check.C) {
> > +	imgName := "bldargtest"
> > +	envKey := "foo"
> > +	envVal := "bar"
> > +	args := []string{
> > +		"--build-arg", fmt.Sprintf("%s=%s", envKey, envVal),
> > +	}
> > +	dockerfile := fmt.Sprintf(`FROM busybox
> > +		ARG %s=
>
> awesome folks! looks we are agreeing on following syntax option:
>
> allow ARG key to define but not assign a default value and treat ARG key=
> same as ARG key="" i.e. define and set default value to ""
>
> Please correct me if didn't get it right. Else I will make the needed
> changes and push updated patch in a bit.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/15182/files#r39179735>.
>